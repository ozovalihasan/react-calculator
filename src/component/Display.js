import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ total, next, operation }) => (
  <div className="display main">
    <div className="display w-100">
      {total || next ? total + operation + next : 0}
    </div>
  </div>
);

Display.propTypes = {
  total: PropTypes.string,
  next: PropTypes.string,
  operation: PropTypes.string,
};

Display.defaultProps = {
  total: '',
  next: '',
  operation: '',
};

export default Display;
