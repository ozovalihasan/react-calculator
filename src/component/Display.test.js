import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import Display from './Display';

let container = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('shows 0 if all props equal to empty string', () => {
  act(() => {
    render(<Display next="" total="" operation="" />, container);
  });
  expect(container.textContent).toBe('0');
});

it('shows joined form of all props if any props is not equal to empty string ', () => {
  act(() => {
    render(<Display next="" total="2" operation="+" />, container);
  });
  expect(container.textContent).toBe('2+');
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Display next="" total="2" operation="+" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
