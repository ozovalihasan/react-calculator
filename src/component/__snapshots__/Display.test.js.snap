// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="display main"
>
  <div
    className="display w-100"
  >
    2+
  </div>
</div>
`;
